dwave/gate/simulator/simulator.pyx

```diff
@@ -26,9 +26,10 @@ import warnings
 from typing import List, Optional
 
 import numpy as np
-
 cimport numpy as np
 
+np.import_array()
```

Comment:
FWIW, I ended up not needing this for `dwave-samplers` and `dwave-preprocessing`. I am not 100% sure in what cases it's necessary or not.

Reply:
Sure, not 100 if it's required either. I went with the recommendation from the [Cython docs](https://cython.readthedocs.io/en/latest/src/tutorial/numpy.html#adding-types):

> Therefore we recommend always calling "import_array" whenever you "cimport numpy"

Reply:
I am less convinced. If testing shows no issue then IMO better not to have a redundant call. It's just another surface for backwards compatibility. And in this code we're not using the NumPy API directly as far as I can tell.

Reply:
I have no strong feelings, so I'm happy to remove it here if all seems to work without it.